### PR TITLE
Use `__self__` instead of `im_self` for Python 3 compatibility

### PR DESCRIPTION
**Pull request by vangheem**

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -4,6 +4,9 @@ Products.CMFCore Changelog
 2.4.0b6 (unreleased)
 --------------------
 
+- Check for `__self__` as well as `im_self` for Python 3 compatibility
```
Comment (ale-rt) [inline CHANGES.rst]: Replacing `im_self` with `__self__` should be already compatible with Python >= 2.7.

Comment (vangheem) [reply CHANGES.rst]: Oh really, even better. So in Python 2.7 `im_self` is alias to `__self__`?

Comment (ale-rt) [reply CHANGES.rst]: See https://docs.python.org/2/reference/datamodel.html: 

> Changed in version 2.6: For Python 3 forward-compatibility, im_func is also available as __func__, and im_self as __self__.

Apparently this alias was added in Python 2.6 #TIL

File: Products/CMFCore/utils.py [hunk 2]
```diff
@@ -238,8 +238,8 @@ def _mergedLocalRoles(object):
             object = object.aq_parent
             object = getattr(object, 'aq_inner', object)
             continue
-        if hasattr(object, 'im_self'):
-            object = object.im_self
+        if hasattr(object, '__self__'):
+            object = object.__self__
```
[the remark this answers is not in the document]
Comment (vangheem) [reply Products/CMFCore/utils.py]: Sorry, updated title. Originally I had a check for both but `__self__` also works in python > 2.6 so it was unnecessary.

[the remark this answers is not in the document]
Comment (tseaver) [reply Products/CMFCore/utils.py]: Cool, it even works for oldstyle classes on 2.6.

**Comment by tseaver** [comment]
Test failures seem unrelated.

**Comment by icemac** [comment]
Test failures are fixed now but having a test for the code change would still be nice.

**Comment by ale-rt** [comment]
@icemac test added

**Comment by ale-rt** [comment]
Thanks for the quick fix :)

**Comment by icemac** [comment]
This PR is released in https://pypi.org/project/Products.CMFCore/2.4.0b6/

**Pull request merged**